tensorflow/python/keras/callbacks.py

```diff
@@ -2328,6 +2329,10 @@ def _log_weights(self, epoch):
       with summary_ops_v2.always_record_summaries():
         for layer in self.model.layers:
           for weight in layer.weights:
+            # allowed_values from tensorflow/core/ops/compat/ops_history_v2/WriteHistogramSummary.pbtxt
+            if weight.dtype not in [ dtypes.float16, dtypes.bfloat16, dtypes.float32, dtypes.float64, dtypes.int8, dtypes.int16, dtypes.int32, dtypes.uint8, dtypes.int64, dtypes.uint16, dtypes.uint32, dtypes.uint64 ]:
```

Comment:
Is the issue only with bool weights? If so, can we just cast bool to int32?

Reply:
Is this private global going to be mainly in sync with https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/compat/ops_history_v2/WriteHistogramSummary.pbtxt#L25?

Reply:
@omalleyt12 I think that any unsupported dtype has a potential to break the callback (e.g. string), hence the whitelist approach.
The global variable approach would transform into something like
```
_SUPPORTED_HISTOGRAM_DTYPES = ...
...
if weight.dtype not in _SUPPORTED_HISTOGRAM_DTYPES:
  ...
```
Are you okay with the above?

@bhack Yes, exactly.
I'm curious if there is a way to just use exactly the protobuf definition and get rid of any extra global variable altogether?

Reply:
I've [committed](https://github.com/tensorflow/tensorflow/pull/43418/commits/80eca5273f8f631da5afac2bf6aedcc2443d8392) the proposed _SUPPORTED_HISTOGRAM_DTYPES solution. Marking as resolved for now.